android/game/src/main/java/com/farsitel/bazaar/game/GameHubBridge.java

```diff
@@ -0,0 +1,422 @@
+                    connectionState = result;
+                    if (connectionState.status == Status.SUCCESS) {
+                        connectionState.message = "GameHub service connected.";
+                        connectionState.call(callback);
+                    }
+                });
+            }
+        };
+
+        // Bind to bazaar game hub
+        Intent serviceIntent = new Intent("com.farsitel.bazaar.Game.BIND");
+        serviceIntent.setPackage(BAZAAR_PACKAGE_NAME);
+
+        PackageManager pm = context.getPackageManager();
+        List<ResolveInfo> intentServices = pm.queryIntentServices(serviceIntent, 0);
+        if (!intentServices.isEmpty()) {
+            // service available to handle that Intent
+            isBroadcastMode = !context.bindService(serviceIntent, gameHubConnection, Context.BIND_AUTO_CREATE);
+            if (isBroadcastMode) {
+                gameHubBroadcast = new GameHubBroadcast(context, logger);
+            }
+        }
+    }
+
+    public Result areServicesAvailable() {
+        boolean isAvailable = gameHubService != null || gameHubBroadcast != null;
+        return new Result(isAvailable ? Status.SUCCESS : Status.DISCONNECTED, isAvailable ? "" : "Connect to service before!");
+    }
+
+    @Override
+    public void isLogin(Context context, boolean showPrompts, IBroadcastCallback callback) {
+        final Result result = new Result(Status.SUCCESS, "");
+        if (gameHubService == null && gameHubBroadcast == null) {
+            result.status = Status.DISCONNECTED;
+            result.message = "Connect to service before!";
+            callback.call(result);
+            return;
+        }
+
+        if (isBroadcastMode) {
+            gameHubBroadcast.isLogin(callback);
+            return;
+        }
+        executorService.submit(() -> {
+            try {
+                if (gameHubService.isLogin()) {
+                    callback.call(result);
+                    return;
+                }
+                result.status = Status.LOGIN_CAFEBAZAAR;
+                result.message = "Login to Cafebazaar before!";
+            } catch (Exception e) {
+                e.printStackTrace();
+                result.status = Status.FAILURE;
+                result.message = e.getMessage();
+                result.stackTrace = Arrays.toString(e.getStackTrace());
+            }
+            MainThread.run(() -> {
+                callback.call(result);
+                if (showPrompts) {
+                    startActionViewIntent(context, "bazaar://login", BAZAAR_PACKAGE_NAME);
+                }
+            });
+        });
+    }
+
+    public void getTournaments(Activity activity, ITournamentsCallback callback) {
+        logger.logDebug("Call getTournaments");
+
+
+        // Check one of services is connected
+        Result serviceResult = areServicesAvailable();
+        if (serviceResult.status != Status.SUCCESS) {
+            serviceResult.call(callback);
```

Comment:
It's a weird way that the result calls the callback! write a general callback that you need when there is no connection with bazaar client